examples/run/sycl/full_chain_algorithm.hpp

```diff
@@ -82,8 +89,12 @@ class full_chain_algorithm
     /// @name Sub-algorithms used by this full-chain algorithm
     /// @{
 
-    /// Host->Device cell copy algorithm
-    device::container_h2d_copy_alg<cell_container_types> m_host2device;
+    /// The number of cells to put together in each partition.
+    /// Equal to the number of threads in the clusterization kernels.
+    /// Adapt to different GPUs' capabilities.
+    unsigned short m_max_cells_per_partition;
+    /// Partitioning algorithm
+    device::partitioning_algorithm m_partitioning;
```

Comment:
Do we need to have this as a dedicated partitioning algorithm rather than an algorithm that is embedded in the clusterization algorithm? It's not like the partitioning code has any use outside of clusterization.

Reply:
Yeah this was a poor design choice from me. I'll change it up to just being a part of the clusterization alg.

Reply:
Okay, that sounds good.